### DISCUSSION TITLE
change settings in config file or while install

### DISCUSSION THREAD

**Discussion by bebeidon** [rustdesk/rustdesk#703] [General]
Is there a way to activate the "direct IP connection" setting and deactivate the server in the config files in the AppData folder on Win11?

Or is there maybe a way to give the installer a config file as parameter with the settings that should be installed?

**Comment by bebeidon** [comment]
setting the password to a custom password would also be nice in this context.

**Reply by dinger1986** [reply]
Yes that is possible.

Flags are:

--uninstall, --after-install, --before-uninstall, --update, --reinstall, --silent-install, --remove. --service, --server, --import-config and --password

https://github.com/rustdesk/rustdesk/blob/85cd066cd711125dc5eda48ef4464d053e06654d/src/main.rs#L74

**Reply by bebeidon** [reply]
thank you for the reply! so is it possible to activate the direct IP connection from the start like this? i don't see how this is configured in the config so how do i have to put it in the config file to work?
is there a config template with all the options somewhere documented?

**Reply by dinger1986** [reply]
I dont see why not, its install arguments so should be able to run it ie setup.exe --server 192.168.10.1 or whatever

**Reply by bebeidon** [reply]
no, that's not what i mean.
i want to activate the "direct IP connection" and deactivate the server setting. so after install i am only able to connect to IP hosts.
so is there a complete config documented, so i can prepare it for the install for this behaviour? or a documentation of the install parameters? it's not clear to me what the following arguments do:
--after-install, --before-uninstall    | what are they doing
--remove    | what is this doing, there is already uninstall, how is it different?
--service    | what behaviour can i expect here? it does not only install service, the app is still there and i was thinking it installs the service anyway.

**Reply by dinger1986** [reply]
ah I have no idea, sorry appears I have answered the wrong question here

**Reply by bebeidon** [reply]
can anybody provide a config template with all the possible settings included?
how do i activate the "direct IP connection" and deactivate the server setting in the config? i can't find it only in the GUI.

**Reply by koma-xrd** [reply]
thats what i found:
change this option to Y or remove option in config file:

C:\Users\username\AppData\Roaming\RustDesk\config\RustDesk2.toml



[options]
direct-server = 'Y'

**Reply by bebeidon** [reply]
i have found this also by now but it does not work to just edit the config file in the installer version, it will not be user while running and when restarting the config will be new generated with your changes gone.

**Reply by duczz** [reply]
i have the same problem :(

**Comment by Johnnie390** [comment]
Hello all,
where are the config files for rustdesk running on Linux (Debian)?

Regards

Aubrey

**Reply by dinger1986** [reply]
Hello,

They should be in the following places 

```
/home/username/.config/rustdesk/RustDesk2.toml
/root/.config/rustdesk/RustDesk2.toml
```

**Reply by Johnnie390** [reply]
Thanks for the information!

Regards

Aubrey

**Comment by Johnnie390** [comment]
Hello all,

if we wanted to implement Firewall rules for a department (+60) of users to use and be administered by Rustdesk, is there any global file/setting that would allow/enable this?

Any help/tips much appreciated.

Regards,

Aubrey

**Reply by dinger1986** [reply]
Rustdesk isnt an RMM, the firewall rules would need set on the machine. Tactical RMM would do this for you

**Comment by Johnnie390** [comment]
Thanks for the information.

Aubrey

**Comment by elhananjair** [comment]
I really wish this is possible while installing rustdesk on all AD joined computers so that manual interventions are not required.

**Comment by rustdesk** [comment]
#763
